presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -10,39 +16,38 @@ class AnonymizerEngine:
     and replaces the PII entities with the desired transformations.
     """
 
-    # Task: 2672
-    # TODO this needs to be implemented currently a stab.
-    # TODO replace analyze_results with domain results
-    # Notice the document Omri created, it impacts the implementation
     def __init__(
             self,
-            text: str,
-            # TODO change to domain object
-            transformations: List[str],
-            # TODO change to domain object
-            analyze_results: List[str],
+            data: dict,
     ):
-        """
-        Handle text replacement for PIIs with requested transformations.
+        """Handle text replacement for PIIs with requested transformations.
 
-        :param text: The original text we want to replace PIIs in
-        :param transformations: The desired transformations - mapping between PII type
-        and transformation type with relevant params
-        :param analyze_results: The results of the analyzer of PIIs locations and scores
+        :param data: a map which contains the transformations, analyzer_results and text
         """
-        self.transformations = transformations
-        self.analyze_results = analyze_results
-        self.text = text
-        self._end_point = len(text)
+        self._transformations = Transformations(data.get("transformations"))
+        self._analyze_results = data.get("analyzer_results")
+        if self._analyze_results is None:
+            self._analyze_results = AnalyzerResults()
+        self._text = data.get("text")
+        if self._text is None:
+            self._text = ""
+        self._end_point = len(self._text)
 
     def anonymize(self):
         """Anonymize method to anonymize the given text.
 
         :return: the anonymized text
         """
-        # TODO a loop that goes through the analyzer results from END to START! reverse.
-        # TODO for each result, replace the old value with the new value
-        # Make sure we handle partial intersections using the endpoint param
-        # TODO dictionary with the transformations and fields
-        # To map field to its transformation
-        pass
+        analyzer_results = self._analyze_results.to_sorted_set(True)
```

Comment:
Since we define it as a standalone service, me might want to use different terminology, and not mention 'analyzer' at all.

Reply:
Well an analyzer is an anlalyzer not necessarily ours.
It's the results from the analyzer :)
So I don't think it makes  a difference.

Reply:
I mean it can just mask data, not result from other service, not analyzation service... these kind of stuff start a conceptual coupling that lead to code coupling, just my 2 cents

Reply:
You are correct... 
But I think it's a great change, all around. Lets consult on this one with the team and open a use story accordingly wdyt?

Reply:
Yes definitely, it's a product decision even more then engineering.
We tackled the same issue in the REST API ADR, we can continue there with product